message_sender/views.py

```diff
@@ -95,6 +95,35 @@ def get_serializer_class(self):
                 return JunebugInboundSerializer
         return InboundSerializer
 
+    def create(self, request, *args, **kwargs):
+        close_event = False
+        if "channel_data" in request.data:  # Handle message from Junebug
+            if "session_event" in request.data["channel_data"]:
+                if request.data["channel_data"]["session_event"] == "close":
+                    close_event = True
+                    reply_field = "reply_to"
+                    from_field = "from"
+        elif "session_event" in request.data:  # Handle message from Vumi
+            if request.data["session_event"] == "close":
+                close_event = True
+                reply_field = "in_reply_to"
+                from_field = "from_addr"
+
+        if close_event:
+            try:
+                message = Outbound.objects.get(
+                    vumi_message_id=request.data[reply_field])
+            except ObjectDoesNotExist:
+                message = Outbound.objects.filter(
+                    to_addr=request.data[from_field]).last()
```

Comment:
I think we need to sort here? I don't think that `last` guarantees us the the last element based on a timestamp.